02-peer-protocol.md

```diff
@@ -50,10 +67,451 @@ pubkey corresponding to the funding output nothing prevents duplicative channel
 ids.
 
 
+### `channel_id`, v2
+
+For channels established using the v2 protocol, the `channel_id` is the
+SHA256(lesser-revocation-basepoint || greater-revocation-basepoint),
+where the lesser and greater is based off the order of the basepoint. The basepoints are compact DER-encoded public keys.
+
+
+## Interactive Transaction Construction
+
+Interactive transaction construction consists of allowing two peers to share inputs
+and outputs for a transaction. This protocol is the foundation for dual-funded
+channels establishment, splicing, multi-purposes closes, as well as
+multi-party (3+ nodes) transactions.
+
+### Set-Up and Vocabulary
+
+There are two parties to an interactive construction: an `initiator` and a `contributor`.
+The `initiator` is the peer which initiates the protocol, e.g. for channel establishment
+the `initiator` would be the peer which sends `open_channel`; for a close,
+the `initiator` sends `shutdown`.
+
+The protocol that follows makes the following assumptions:
+
+- The `feerate` for the transaction has already been agreed upon.
+- The `nLocktime` for the transaction has already been communicated, or is
+  established via convention.
+- The transaction version is 2.
+- The each input's sequence number will be set to 0xFDFFFFFF (little endian), which signals RBF.
```

Comment:
What about inputs that spend OP_CSV outputs?

Reply:
Allowing inputs that have time-based* spending restrictions means that you'd need to know the 1) block the input's outpoint was included in and 2) the current block-time/block-height in order to know if that's currently spendable.

2) is trivial enough to figure out but for light-clients, 1) is a bit prohibitive.

so by disallowing the sequence number we're remove one vector for producing an un-mineable transaction.

The real question here, then, are we ok allowing the interactive tx protocol to produce un-mineable transactions? I lean towards "no", as this protocol is meant to be general enough to allow for building cooperative close and splicing transactions -- you wouldn't want to have a 100 block wait on a splicing transactions particularly.

The obvious downside to this is that with the introduction of anchor outputs, the following lightning protocol transactions would need to be spent elsewhere before being eligible for use in an interactive tx exchange:
- unilaterally closed tx's to_local_anchor and to_remote_anchor outputs
- unilaterally closed tx's to_remote outputs
- non-revoked htlc-success txs
- non-revoked htlc-timeout txs

*block height or block-time

Reply:
On further consideration, given that the lightning spec itself will now be generating outputs with sequence numbers and that this protocol is intended to be versatile enough to handle several different transaction construction scenarios, it's a good idea to include a field for the sequence number.